src/coreclr/gc/gcconfig.h

```diff
@@ -138,7 +138,7 @@ class GCConfigStringHolder
     INT_CONFIG   (GCWriteBarrier,            "GCWriteBarrier",            NULL,                                0,                  "Specifies whether GC should use more precise but slower write barrier")                  \
     STRING_CONFIG(GCName,                    "GCName",                    "System.GC.Name",                                        "Specifies the path of the standalone GC implementation.")                                \
     INT_CONFIG   (GCSpinCountUnit,           "GCSpinCountUnit",           0,                                   0,                  "Specifies the spin count unit used by the GC.")                                          \
-    BOOL_CONFIG  (GCDynamicAdaptation,       "GCDynamicAdaptation",       NULL,                                false,              "Enables varying the heap count dynamically in Server GC.")
+    BOOL_CONFIG  (GCDynamicAdaptation,       "GCDynamicAdaptation",       "System.GC.DynamicAdaptation",       false,              "Enables varying the heap count dynamically in Server GC.")
```

Comment:
I would probably take the opportunity to change this to an int config in case we'll want more values on this config. 

also I would change the comment to say "Enable the GC to dynamically adapt to application sizes". changing the heap count is merely a mechanism and is not all we do for this config.

Reply:
hmm, should we rename it something like `GCDynamicAdaptionLevel` or something then? Just `GCDynamicAdaption=2` feels little odd

Reply:
yep, that does sound a little odd. maybe `Mode` is more appropriate? as in, adapting to different perf aspects, size is one of them.

Reply:
sure we can go with `GCDynamicAdaptationMode`.

Reply:
updated the config name, and made it INT_CONFIG with explicit checks for `1` for now, if we have additional modes we can define some constants or them.